bouncer/tests/btc_vault_swap.ts

```
@@ -0,0 +1,120 @@
+  chain: string;
+  nulldata_utxo: string;
+  deposit_address: string;
+}
+
+async function buildAndSendBtcVaultSwap(
+  depositAmountBtc: number,
+  brokerUri: string,
+  destinationAsset: Asset,
+  destinationAddress: string,
+  refundAddress: string,
+) {
+  await using chainflip = await getChainflipApi();
+
+  const broker = createStateChainKeypair(brokerUri);
+  testBtcVaultSwap.debugLog('Broker:', broker.address);
+  testBtcVaultSwap.debugLog(`Btc endpoint is set to`, BTC_ENDPOINT);
+
+  const feeBtc = 0.00001;
+  const { inputs, change } = await selectInputs(Number(depositAmountBtc) + feeBtc);
+
+  const vaultSwapDetails = (await chainflip.rpc(
+    `cf_get_vault_swap_details`,
+    broker.address,
+    'BTC', // source_asset
+    destinationAsset.toUpperCase(),
+    destinationAddress,
+    0, // broker_commission
+    0, // min_output_amount
+    0, // retry_duration
+  )) as unknown as VaultSwapDetails;
+
+  assert.strictEqual(vaultSwapDetails.chain, 'Bitcoin');
+  testBtcVaultSwap.debugLog('nulldata_utxo:', vaultSwapDetails.nulldata_utxo);
+
+  // The `createRawTransaction` function will add the op codes, so we have to remove them here.
+  const nullDataWithoutOpCodes = vaultSwapDetails.nulldata_utxo.replace('0x', '').substring(4);
+
+  const outputs = [
+    {
+      [vaultSwapDetails.deposit_address]: depositAmountBtc,
+    },
+    {
+      data: nullDataWithoutOpCodes,
+    },
+    {
+      [refundAddress]: change,
+    },
+  ];
+
+  const rawTx = await btcClient.createRawTransaction(inputs, outputs, 0, false);
+  const signedTx = await btcClient.signRawTransactionWithWallet(rawTx);
+  const txid = await btcClientMutex.runExclusive(async () =>
+    btcClient.sendRawTransaction(signedTx.hex),
+  );
+  if (!txid) {
+    throw new Error('Broadcast failed');
+  } else {
+    testBtcVaultSwap.log('Broadcast successful, txid:', txid);
+  }
+
+  await waitForBtcTransaction(txid as string);
+  testBtcVaultSwap.debugLog('Transaction confirmed');
```

This isn't necessarily true. 

We either exit `waitForBtcTransaction` early because we have the confirmations, or we exit it later because it hasn't been confirmed. But we'll still print confirmed here. 

`waitForBtcTransaction` should throw an error if it's not found.